### ISSUE TITLE
[influxdb] historic states of switch items are always OFF with influxdb v1

### ISSUE THREAD

**Issue by olivierkeke** [openhab/openhab-addons#9455]
With an influxdb v1 database, OH3 always read historical historical state of switch items as `OFF`. Yet, values are well stored in the database (can be seen with grafana). 
These community posts are related to this problem:
  - https://community.openhab.org/t/oh3-restoreonstartup-not-working/110592
  - https://community.openhab.org/t/oh3-graph-of-switch-item/109811.
  - https://community.openhab.org/t/oh3-rule-check-in-a-rule-if-item-was-on-in-past-30-minutes/110567

**Comment by servim** [comment]
I have the same problem with Influx in both [v1](https://community.openhab.org/t/guide-installing-and-testing-oh3-snapshots-milestones-using-apt-and-yum/104572/140?u=olymp) and [v2](https://community.openhab.org/t/habpanel-timeline-influxdb2-oh3/109016?u=olymp).

**Comment by xsirpreisx** [comment]
Yes, same problem here.

**Comment by t2000** [comment]
I found the culprit, it is here:

https://github.com/openhab/openhab-addons/blob/8cbd358b848b1b47539f36609bd7bd40167e5134/bundles/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBStateConvertUtils.java#L147-L151

Inside the influxdb there is "1.0" instead of "1" stored (probably from the OH-1 binding in our cases). And thus:

https://github.com/openhab/openhab-addons/blob/8cbd358b848b1b47539f36609bd7bd40167e5134/bundles/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBStateConvertUtils.java#L151

will be executed and this returns `false` for `object = "1.0"` :)

**Comment by xsirpreisx** [comment]
Perfect, thank you. When will be able to use that fix? Do I have to update to the latest master to get it and wait for the next stable release if I want to stay on the stable branch?

**Comment by t2000** [comment]
@xsirpreisx This PR is not part of the master branch, its still waiting to be merged into master.

After that has happened (you see it when the PR is marked as "merged"), there are nightly build jobs on ci.openhab.org which build snapshot releases, so in the following night you should be able to download a new OH version from there.

However, please be careful with those snapshots as they might be unstable... (hence SNAPSHOT, not release ;)

If you want to stay on a stable version you will most likely have to wait for at least 3.1 (or later, depending on when this PR gets merged).

**Comment by xsirpreisx** [comment]
Thanks. Will I be getting the latest snapshot builds if I change my branch in openhabian-config to master? Or is that even another kind of branch?

3.1 is too far away for me to wait. restoreOnStartup is quite an important feature for me (and I'd also like to use changedSince and other persistence stuff), and I don't really want to use a different persistance database either...

**Comment by t2000** [comment]
I do not use openhabian, but there is probably a way to tell openhabian to use either stable or snapshots (note: the PR to fix this issue is NOT merged into ANY official branch yet!).

But: You have been warned! Snapshots MIGHT contain broken stuff.

**Comment by xsirpreisx** [comment]
I understand. I was using snapshots before, but that was a few years ago.

And for me, the stable 3.0 is broken right now, so I might try  snapshot anyway, once this fix is in there ;-)

**Comment by servim** [comment]
i just upgraded to 3.1 2105
my problem was not solved.
I will try to briefly tell you the essence of one of the problems described in the topics, the links to which I gave above.
I am using HabPanel + Influx v2.
In the HabPanel, I use a timeline widget, which should display the state of an item of type OnOff for a period.
In chrome browser developer tools i am getting errors when updating widget.
request:
```
http://192.168.215.231:8080/rest/persistence/items/Box_HeaterIR?boundary=true&starttime=2020-12-26T02:21:23.189Z&serviceId=influxdb
```
result:
```
"{\"error\":{\"message\":\"error in building plan while starting program: cannot query an empty range\",\"http-code\":500,\"exception\":{\"class\":\"com.influxdb.exceptions.BadRequestException\",\"message\":\"error in building plan while starting program: cannot query an empty range\",\"localized-message\":\"error in building plan while starting program: cannot query an empty range\"}}}"
```
if you exclude the parameter in the request - boundary=true
```
http://192.168.215.231:8080/rest/persistence/items/Box_HeaterIR?starttime=2020-12-26T02:21:23.189Z&serviceId=influxdb
```
then the result will be like this:
```
{"name":"Box_HeaterIR","datapoints":"11","data":[{"time":1608976150615,"state":"OFF"},{"time":1609032596100,"state":"OFF"},{"time":1609032596100,"state":"ON"},{"time":1609063184354,"state":"ON"},{"time":1609063184354,"state":"OFF"},{"time":1609121546486,"state":"OFF"},{"time":1609121546486,"state":"ON"},{"time":1609148710583,"state":"ON"},{"time":1609148710583,"state":"OFF"},{"time":1609199316786,"state":"OFF"},{"time":1609199316786,"state":"ON"}]}
```
already better, but I don't understand why each state is returned 2 times in a row: ON, ON, OFF, OFF
Thanks!

**Comment by t2000** [comment]
>i just upgraded to 3.1 2105
my problem was not solved.

@servim What do you NOT understand by me saying, this is ***NOT*** merged?

**Comment by servim** [comment]
> What do you NOT understand

I apologize! My English is very bad, I noticed another post:

> so in the following night you should be able to download a new OH version from there.

Sorry again, I was not attentive.


> please do not pollute this issue with it, thank you.

I am not an expert to distinguish that these are different problems and, just in case, drew attention to this, suddenly it is important.
PS I was just trying to help by providing information, and not sitting and waiting for someone to find the problem, someone to report them, and someone to solve them.
I'm sorry if I offended you with something :(

**Comment by t2000** [comment]
@servim No offense taken.

It just looks like that several people are waiting for this fix. All I can say is: The fix is done, but not yet merged. Everyone working on openHAB is doing this in their spare time, so everything will be done when its done.

Please open a new issue for your problem, because it is clearly unrelated from the fact that the addon currently reads out "1.0" as false.

**Comment by olivierkeke** [comment]
@servim @xsirpreisx If you want to stay with stable release, you can build the influxdb binding from the main branch and put the generated jar in your `addons` directory.

**Comment by servim** [comment]
> Please open a new issue for your problem

ОК, thanks, I will do the second version of the Influx, and the first one will wait for the picture, maybe this will fix my problem and mine too.


> with stable release

I'm only in pictures, so I'll wait for the desired picture, thanks!

**Comment by xsirpreisx** [comment]
> @servim @xsirpreisx If you want to stay with stable release, you can build the influxdb binding from the main branch and put the generated jar in your `addons` directory.

Perfect, thanks. The fix seems to be working!

**Comment by lsiepel** [comment]
It seems to be merged now. But it's merged against 3.0.x shouldn't it be merged against master?

**Comment by olivierkeke** [comment]
> It seems to be merged now. But it's merged against 3.0.x shouldn't it be merged against master?

It is merged against `main` since 29 Dec 2020.
